### PR TITLE
Mailchimp integration

### PR DESCRIPTION
**Pull request by pettinarip**
### Description
Integration with Mailchimp when a user signup to the newsletter.

- I have also created a `.env.local.example` file to have a reference of all the env vars that we have in the app.

### TODO
- [x] test the integration using the EF mailchimp api key. @samajammin to provide one

### CODE REVIEW

File: src/components/forms/NewsletterSignup.tsx [hunk 1]
```diff
@@ -35,29 +47,42 @@ export const NewsletterSignup: FC = () => {
...
-        containerStyle: {
-          fontFamily: 'fonts.heading'
+      try {
+        const response = await fetch(API_NEWSLETTER_SIGNUP_URL, {
+          method: 'POST',
+          body: data.email
+        });
+
+        if (!response.ok) {
+          const errorText = await response.text();
```
[the remark this answers is not in the document]
Comment (nhsz) [reply src/components/forms/NewsletterSignup.tsx:65]: Nice!

File: src/pages/api/newsletter-signup.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+mailchimp.setConfig({
+  apiKey: process.env.MAILCHIMP_APIKEY,
+  server: process.env.MAILCHIMP_SERVER
+});
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { body, method } = req;
+
+  if (method !== 'POST') {
```
Comment (nhsz) [inline src/pages/api/newsletter-signup.ts:12]: Is this necessary?

**Review by nhsz — APPROVED** [review]
LGTM! 👍 Left a couple of comments.

File: src/components/forms/NewsletterSignup.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (pettinarip) [reply src/components/forms/NewsletterSignup.tsx:65]: Yes, I was trying to look into the docs but I couldn't find a list of them.
Perhaps it is better to:
- remove this and just display a generic error.
- or only display the error messages for 400 errors https://mailchimp.com/developer/marketing/docs/errors/

What do you think?

File: src/components/forms/NewsletterSignup.tsx [hunk 3]
```diff
@@ -35,29 +47,42 @@ export const NewsletterSignup: FC = () => {
...
-      toast({
-        position: 'top-right',
-        title: 'Success!',
-        status: 'success',
-        duration: 5000,
-        isClosable: true,
-        containerStyle: {
-          fontFamily: 'fonts.heading'
+      try {
+        const response = await fetch(API_NEWSLETTER_SIGNUP_URL, {
```
[the remark this answers is not in the document]
Comment (pettinarip) [reply src/components/forms/NewsletterSignup.tsx:59]: I like that! yes, once you get that merge we can refactor this. I couldn't fully test it yet so probably you will merge that code first and we are gonna be able to refactor this piece of code in this PR. Will see.

File: src/pages/api/newsletter-signup.ts (hunk 2 above)
Comment (pettinarip) [reply src/pages/api/newsletter-signup.ts:12]: Hmm not totally necessary I would say, just a guard check, to fail quickly. Let me know if you feel this is not adding much value and I can remove it.

File: src/components/forms/NewsletterSignup.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nhsz) [reply src/components/forms/NewsletterSignup.tsx:65]: I'd go with displaying the mailchimp error messages for 400 errors and a generic message for the rest, just to be safe.

File: src/pages/api/newsletter-signup.ts (hunk 2 above)
Comment (nhsz) [reply src/pages/api/newsletter-signup.ts:12]: Not sure, I've never use a guard like that to be honest, don't know if the method can be 'hijacked' someway. What do you think @samajammin?

**Comment by pettinarip** [comment]
Tested successfully with an EF account. Merging it. @samajammin feel free to add comments on the open discussions and we can open a new PR in case its needed.

**Pull request merged**